mne/bem.py

```diff
@@ -827,7 +837,32 @@ def fit_sphere_to_headshape(info, dig_kinds=(FIFF.FIFFV_POINT_EXTRA,),
     This function excludes any points that are low and frontal
     (``z < 0 and y > 0``) to improve the fit.
     """
-    # get head digization points of the specified kind
+    if units is None:
+        warn('Please explicitly set the units. In 0.12 units="mm" will '
+             'be used, but this will change to units="m" in 0.13.',
+             DeprecationWarning)
+        units = 'mm'
+    if not isinstance(units, string_types) or units not in ('m', 'mm'):
+        raise ValueError('units must be a "m" or "mm"')
+    if isinstance(dig_kinds, string_types):
+        if dig_kinds == 'auto':
+            # try "extra" first
+            try:
+                return fit_sphere_to_headshape(info, 'extra', units=units)
+            except ValueError:
+                pass
```

Comment:
On Py3k if you put something that raises an Exception in a `except` like this:
```
def fail():
    raise RuntimeError()


try:
    fail()
except RuntimeError:
    fail()
```
You get this excessively long traceback:
```
Traceback (most recent call last):
  File "C:\Users\larsoner\Desktop\untitled1.py", line 13, in <module>
    fail()
  File "C:\Users\larsoner\Desktop\untitled1.py", line 9, in fail
    raise RuntimeError()
RuntimeError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\larsoner\Desktop\untitled1.py", line 15, in <module>
    fail()
  File "C:\Users\larsoner\Desktop\untitled1.py", line 9, in fail
    raise RuntimeError()
RuntimeError
```
Whereas if you do:
```
try:
    fail()
except RuntimeError:
    pass
fail()
```
You get:
```
Traceback (most recent call last):
  File "C:\Users\larsoner\Desktop\untitled1.py", line 16, in <module>
    fail()
  File "C:\Users\larsoner\Desktop\untitled1.py", line 9, in fail
    raise RuntimeError()
RuntimeError
```
Here it's entirely possible the next call to `fit_sphere_to_headshape` to fail so it's better to put it outside the `except`.